### PR TITLE
Import thread utils from Marathon.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
The fixed named thread pool and retry can be used by Cosmos and others.

### CODE REVIEW

File: commons/src/main/scala/com/mesosphere/usi/async/Retry.scala [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    maxAttempts: Int = Retry.DefaultMaxAttempts,
+    minDelay: Duration = Retry.DefaultMinDelay,
+    maxDelay: Duration = Retry.DefaultMaxDelay,
+    maxDuration: Duration = Retry.DefaultMaxDuration
+)
+
+object RetryConfig {
+  def apply(config: Config): RetryConfig = {
+    RetryConfig(
+      config.int("max-attempts", default = Retry.DefaultMaxAttempts), // TODO: should we import RichConfig?
```
Comment (jeschkies) [inline commons/src/main/scala/com/mesosphere/usi/async/Retry.scala]: I wonder if we should import https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/util/RichConfig.scala

File: commons/src/main/scala/com/mesosphere/usi/async/ThreadPoolContext.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.mesosphere.usi.async
+
+import java.util.concurrent.{ExecutorService, Executors, ThreadFactory}
+// TODO: Should I import Guava?
+import com.google.common.util.concurrent.ThreadFactoryBuilder
```
Comment (jeschkies) [inline commons/src/main/scala/com/mesosphere/usi/async/ThreadPoolContext.scala:4]: @timcharper, is there another elegant way to create a thread poll factory?

**Comment by timcharper** [comment]
Exit status 1 when running "sbt +test:compile"

<details>
<summary>Details</summary>
<pre>
[info] Compiling 3 Scala sources to /home/travis/build/mesosphere/usi/persistence-zookeeper/target/scala-2.12/classes ...
[info] Done compiling.
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/mesos-master-detector/target/scala-2.12/classes ...
[info] Done compiling.
[warn] there were three deprecation warnings (since 2.6.0); re-run with -deprecation for details
[warn] one warning found
[info] Done compiling.
[info] Compiling 3 Scala sources to /home/travis/build/mesosphere/usi/persistence-zookeeper/target/scala-2.12/test-classes ...
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/mesos-master-detector/target/scala-2.12/test-classes ...
[warn] /home/travis/build/mesosphere/usi/persistence-zookeeper/src/test/scala/com/mesosphere/usi/storage/zookeeper/ZooKeeperPersistenceStoreTest.scala:37:21: parameter value i in value $anonfun is never used
[warn]     (1 to size).map(i =&gt; Node(randomPath(prefix), ByteString(payload)))
[warn]                     ^
[info] Done compiling.
[warn] there were three deprecation warnings (since 2.6.0); re-run with -deprecation for details
[warn] two warnings found
[info] Done compiling.
[error] (commons / Compile / compileIncremental) Compilation failed
[error] Total time: 40 s, completed Jun 24, 2020 3:25:46 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Unused imports / scalafix"

<details>
<summary>Details</summary>
<pre>
[info] Done compiling.
[info] Running scalafix on 3 Scala sources
[info] Compiling 3 Scala sources to /home/travis/build/mesosphere/usi/persistence-zookeeper/target/scala-2.13/classes ...
[info] Compiling 15 Scala sources to /home/travis/build/mesosphere/usi/test-utils/target/scala-2.13/classes ...
[warn] there were 14 deprecation warnings (since 2.13.0); re-run with -deprecation for details
[warn] one warning found
[info] Done compiling.
[info] Running scalafix on 3 Scala sources
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/mesos-master-detector/target/scala-2.13/classes ...
[info] Done compiling.
[warn] there were 5 deprecation warnings (since 2.13.0)
[warn] there were three deprecation warnings (since 2.6.0)
[warn] there were 8 deprecation warnings in total; re-run with -deprecation for details
[warn] three warnings found
[info] Done compiling.
[info] Running scalafix on 15 Scala sources
[error] (commons / Compile / compileIncremental) Compilation failed
[error] Total time: 37 s, completed Jun 24, 2020 3:27:03 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Running tests for all Scala versions"

<details>
<summary>Details</summary>
<pre>
[info]   + Then we have the hostname and port 
[info] The master detector
[info] - should detect the Mesos master
[info]   + Given the Zookeeper server at zk://127.0.0.1:32770/mesos and a detector 
[info]   + When we detect the Mesos master 
[info]   + Then the URL is the same as our master 
[info] Run completed in 10 seconds, 165 milliseconds.
[info] Total number of tests run: 33
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 33, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Run completed in 9 seconds, 68 milliseconds.
[info] Total number of tests run: 8
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 8, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[error] (commons / Compile / compileIncremental) Compilation failed
[error] Total time: 22 s, completed Jun 24, 2020 3:27:42 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Building documentation"

<details>
<summary>Details</summary>
<pre>
[error]       config.duration(&quot;max-duration&quot;, default = Retry.DefaultMaxDuration)
[error]                                       ^
[error] /home/travis/build/mesosphere/usi/commons/src/main/scala/com/mesosphere/usi/async/Retry.scala:100:13: not found: value scheduler
[error]             scheduler.scheduleOnce(nextDelay)(retry(attempt + 1, nextDelay))
[error]             ^
[error] /home/travis/build/mesosphere/usi/commons/src/main/scala/com/mesosphere/usi/async/ThreadPoolContext.scala:5:12: object google is not a member of package com
[error] import com.google.common.util.concurrent.ThreadFactoryBuilder
[error]            ^
[error] /home/travis/build/mesosphere/usi/commons/src/main/scala/com/mesosphere/usi/async/ThreadPoolContext.scala:32:38: not found: type ThreadFactoryBuilder
[error]     val factory: ThreadFactory = new ThreadFactoryBuilder().setNameFormat(s&quot;$namePrefix-thread-%d&quot;).build()
[error]                                      ^
[warn] /home/travis/build/mesosphere/usi/commons/src/main/scala/com/mesosphere/usi/async/ThreadPoolContext.scala:5:42: Unused import
[warn] import com.google.common.util.concurrent.ThreadFactoryBuilder
[warn]                                          ^
[warn] one warning found
[error] 12 errors found
[error] (commons / Compile / compileIncremental) Compilation failed
[error] Total time: 6 s, completed Jun 24, 2020 3:28:05 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

File: commons/src/main/scala/com/mesosphere/usi/async/ThreadPoolContext.scala (hunk 2 above)
Comment (timcharper) [reply commons/src/main/scala/com/mesosphere/usi/async/ThreadPoolContext.scala:4]: Not sure, I haven't done this much

**Review by timcharper — APPROVED** [review]

**Comment by timcharper** [comment]
Exit status 1 when running "sbt +test:compile"

<details>
<summary>Details</summary>
<pre>
[info] Compiling 3 Scala sources to /home/travis/build/mesosphere/usi/persistence-zookeeper/target/scala-2.12/classes ...
[info] Done compiling.
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/mesos-master-detector/target/scala-2.12/classes ...
[info] Done compiling.
[warn] there were three deprecation warnings (since 2.6.0); re-run with -deprecation for details
[warn] one warning found
[info] Done compiling.
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/mesos-master-detector/target/scala-2.12/test-classes ...
[info] Compiling 3 Scala sources to /home/travis/build/mesosphere/usi/persistence-zookeeper/target/scala-2.12/test-classes ...
[warn] /home/travis/build/mesosphere/usi/persistence-zookeeper/src/test/scala/com/mesosphere/usi/storage/zookeeper/ZooKeeperPersistenceStoreTest.scala:37:21: parameter value i in value $anonfun is never used
[warn]     (1 to size).map(i =&gt; Node(randomPath(prefix), ByteString(payload)))
[warn]                     ^
[info] Done compiling.
[warn] there were three deprecation warnings (since 2.6.0); re-run with -deprecation for details
[warn] two warnings found
[info] Done compiling.
[error] (commons / Compile / compileIncremental) Compilation failed
[error] Total time: 45 s, completed Jun 25, 2020 10:51:39 AM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Running tests for all Scala versions"

<details>
<summary>Details</summary>
<pre>
[info]   + Then we have the hostname and port 
[info] The master detector
[info] - should detect the Mesos master
[info]   + Given the Zookeeper server at zk://127.0.0.1:32770/mesos and a detector 
[info]   + When we detect the Mesos master 
[info]   + Then the URL is the same as our master 
[info] Run completed in 9 seconds, 628 milliseconds.
[info] Total number of tests run: 8
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 8, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Run completed in 10 seconds, 548 milliseconds.
[info] Total number of tests run: 33
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 33, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[error] (commons / Compile / compileIncremental) Compilation failed
[error] Total time: 23 s, completed Jun 25, 2020 10:54:26 AM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Pull request merged**